delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/indexing/ElasticSearchGlobalEventLog.scala

```diff
@@ -55,20 +54,23 @@ class ElasticSearchGlobalEventLog private (
       .map(_.toMessage)
       .groupWithin(batchMaxSize, batchMaxTimeout)
       .discardDuplicates()
-      .evalMapFilterValue(event =>
-        eventExchanges.findFor(event).traverse { ee =>
-          for {
-            state  <- ee.toState(event, tag)
-            graph  <- state.toGraph
-            source <- state.toSource
-          } yield graph.map { g =>
-            IndexingData(
-              g.filter { case (s, p, _) => s == subject(state.state.id) && graphPredicates.contains(p) },
-              source.value
-            )
+      .evalMapFilterValue { event =>
+        Stream
+          .fromIterator[Task](referenceExchanges.iterator)
+          .evalMap { _(event, tag) }
+          .collect { case Some(value) => value }
```

Comment:
shouldn't it be collectFirst? and why are you using a stream at all? Can't you just do a `referenceExchanges.iterator.foldLeftM` instead of all this stream overhead?